src/Internal/Wallet/Cip30Mock.purs

```diff
@@ -206,3 +242,30 @@ mkCip30Mock pKey mSKey = do
 
 -- returns an action that removes the mock.
 foreign import injectCip30Mock :: String -> Cip30Mock -> Effect (Effect Unit)
+
+-- Helpers
+
+foreign import raisePaginateError :: forall a. Int -> Effect a
+
+foreign import raiseAPIError :: forall a. Effect a
+
+paginateArray :: forall a. Maybe Paginate -> Array a -> Effect (Array a)
+paginateArray (Just { page, limit }) xs = process $ pages !! page
+  where
+  chunkBy :: forall a. Int -> Array a -> Array (Array a)
```

Comment:
This is O(n^2). Since there may be thousands of utxos, it is a problem. Do not use `drop`&`take` (it's `O(n+m)`, where `n` and `m` are lengths of the resulting arrays), recursively iterate over `Int` indices instead and use `Array.slice`. That way each element of the original array will "participate" only in 1 `slice` call.